From b4c1cc7e4d32c6aaa2d2466b3833318d6ef647ba Mon Sep 17 00:00:00 2001
From: Florin Sarbu <florin@resin.io>
Date: Wed, 9 May 2018 14:19:58 +0200
Subject: [PATCH] artik_common.h: Rework the machine specific integration of
 the resin-u-boot environment configuration

Because the kernel with the bundled initramfs has grown in size, we need to allow
for some space between the addresses where the kernel and device tree are loaded
in u-boot so they do not overlap. Without this increase, the kernel will silently
fail to boot. We choose to increase the address where the device tree gets loaded.

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Florin Sarbu <florin@resin.io>
---
 include/configs/artik_common.h | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/include/configs/artik_common.h b/include/configs/artik_common.h
index 9aeb14c..77b52f8 100644
--- a/include/configs/artik_common.h
+++ b/include/configs/artik_common.h
@@ -289,7 +289,7 @@
 	"fdtfile=" CONFIG_FDT_FILE "\0"					\
 	"kernel_file=zImage\0"						\
 	"kernel_addr=40008000\0"					\
-	"fdtaddr=40800000\0"						\
+	"fdtaddr=44800000\0"						\
 	"initrd_file=uInitrd\0"						\
 	"initrd_addr=43000000\0"					\
 	"sdrecovery=sdfuse format; sdfuse flashall 3\0"			\
@@ -298,8 +298,8 @@
 	"factory_set_ethaddr=run factory_load; gen_eth_addr ;"		\
 		"factory_info write ethaddr $ethaddr;"			\
 		"run factory_save\0"					\
-	"boot_cmd=fatload mmc 0:1 $kernel_addr $kernel_file;"		\
-		"fatload mmc 0:1 $fdtaddr $fdtfile;"			\
+	"boot_cmd=fatload mmc ${resin_dev_index}:${resin_boot_part} $kernel_addr $kernel_file;"		\
+		"fatload mmc ${resin_dev_index}:${resin_boot_part} $fdtaddr $fdtfile;"			\
 		"bootz $kernel_addr - $fdtaddr\0"			\
 	"boot_cmd_initrd=fatload mmc 0:1 $kernel_addr $kernel_file;"	\
 		"fatload mmc 0:1 $fdtaddr $fdtfile;"			\
@@ -322,8 +322,18 @@
 		"asix.macaddr=${ethaddr} bd_addr=${bd_addr};"		\
 		"run boot_cmd_initrd\0"					\
 	"mmcboot=setenv bootargs ${console} "				\
-		"root=/dev/mmcblk${rootdev}p${rootpart} ${root_rw} "	\
+		"${resin_kernel_root} ${root_rw} "			\
 		"rootfstype=ext4 ${opts};run boot_cmd\0"		\
-	"bootcmd=run ramfsboot\0"
+	"bootcmd=setenv resin_kernel_load_addr ${kernel_addr}; "	\
+		"run resin_set_kernel_root; "				\
+		"if test -n \"${resin_flasher_dev_index}\"; "		\
+			"then sdfuse format; "				\
+			"sdfuse flash 1 fwbl1 bl1.bin; "		\
+			"sdfuse flash 1 bl2 bl2.bin; "			\
+			"sdfuse flash 1 bootloader u-boot.bin; "	\
+			"sdfuse flash 1 tzsw tzsw.bin; "		\
+			"setenv opts ${opts} rootdelay=3; "		\
+		"fi; "							\
+		"run mmcboot\0"
 
 #endif /* __ARTIK_COMMON_H */
-- 
2.7.4

